pkg/user/RoleGroupService.go

```diff
@@ -525,3 +527,44 @@ func (impl RoleGroupServiceImpl) DeleteRoleGroup(bean *bean.RoleGroup) (bool, er
 
 	return true, nil
 }
+
+func (impl RoleGroupServiceImpl) FetchRolesForGroups(groupNames []string) ([]*bean.RoleFilter, error) {
+	if len(groupNames) == 0 {
+		return nil, nil
+	}
+	roleGroups, err := impl.roleGroupRepository.GetRoleGroupListByNames(groupNames)
+	if err != nil && err != pg.ErrNoRows {
+		impl.logger.Errorw("error while fetching user from db", "error", err)
+		return nil, err
+	}
+	if err == pg.ErrNoRows {
+		impl.logger.Warnw("no result found for role groups", "groups", groupNames)
+		return nil, nil
```

Comment:
should it not return empty list instead of nil?

Reply:
nil handled in rest handler, and this service is not exposed to user only used for internal auth.